common/testData/localeIdentifiers/localeCanonicalization.txt

```diff
@@ -0,0 +1,1647 @@
+aaa_Adlm_FX_fonipa	;	aaa_Adlm_FR_fonipa
+aaa_Adlm_HV_fonipa	;	aaa_Adlm_BF_fonipa
+aaa_Adlm_JT_fonipa	;	aaa_Adlm_UM_fonipa
+aaa_Adlm_MI_fonipa	;	aaa_Adlm_UM_fonipa
+aaa_Adlm_NH_fonipa	;	aaa_Adlm_VU_fonipa
+aaa_Adlm_NQ_fonipa	;	aaa_Adlm_AQ_fonipa
+aaa_Adlm_NT_fonipa	;	aaa_Adlm_SA_fonipa
+aaa_Adlm_PC_fonipa	;	aaa_Adlm_FM_fonipa
+aaa_Adlm_PU_fonipa	;	aaa_Adlm_UM_fonipa
+aaa_Adlm_PZ_fonipa	;	aaa_Adlm_PA_fonipa
+aaa_Adlm_QU_fonipa	;	aaa_Adlm_EU_fonipa
+aaa_Adlm_RH_fonipa	;	aaa_Adlm_ZW_fonipa
+aaa_Adlm_SU_fonipa	;	aaa_Adlm_RU_fonipa
+aaa_Adlm_TP_fonipa	;	aaa_Adlm_TL_fonipa
+aaa_Adlm_UK_fonipa	;	aaa_Adlm_GB_fonipa
+aaa_Adlm_VD_fonipa	;	aaa_Adlm_VN_fonipa
+aaa_Adlm_WK_fonipa	;	aaa_Adlm_UM_fonipa
+aaa_Adlm_YD_fonipa	;	aaa_Adlm_YE_fonipa
+aaa_Adlm_YU_fonipa	;	aaa_Adlm_RS_fonipa
+aaa_Adlm_ZR_fonipa	;	aaa_Adlm_CD_fonipa
+aaa_Qaai_AC_fonipa	;	aaa_Zinh_AC_fonipa
+aam_Adlm_AC_fonipa	;	aas_Adlm_AC_fonipa
+aar_Adlm_AC_fonipa	;	aa_Adlm_AC_fonipa
+aar_Adlm_AC_fonipa_saaho	;	ssy_Adlm_AC_fonipa
+abk_Adlm_AC_fonipa	;	ab_Adlm_AC_fonipa
+adp_Adlm_AC_fonipa	;	dz_Adlm_AC_fonipa
+afr_Adlm_AC_fonipa	;	af_Adlm_AC_fonipa
+aju_Adlm_AC_fonipa	;	jrb_Adlm_AC_fonipa
+aka_Adlm_AC_fonipa	;	ak_Adlm_AC_fonipa
+alb_Adlm_AC_fonipa	;	sq_Adlm_AC_fonipa
+als_Adlm_AC_fonipa	;	sq_Adlm_AC_fonipa
+amh_Adlm_AC_fonipa	;	am_Adlm_AC_fonipa
+ara_Adlm_AC_fonipa	;	ar_Adlm_AC_fonipa
+arb_Adlm_AC_fonipa	;	ar_Adlm_AC_fonipa
+arg_Adlm_AC_fonipa	;	an_Adlm_AC_fonipa
+arm_Adlm_AC_arevela_arevmda_fonipa	;	hyw_Adlm_AC_fonipa
+arm_Adlm_AC_arevela_fonipa	;	hy_Adlm_AC_fonipa
+arm_Adlm_AC_arevmda_fonipa	;	hyw_Adlm_AC_fonipa
+arm_Adlm_AC_fonipa	;	hy_Adlm_AC_fonipa
+art_Adlm_AC_fonipa_lojban	;	jbo_Adlm_AC_fonipa
+asd_Adlm_AC_fonipa	;	snz_Adlm_AC_fonipa
+asm_Adlm_AC_fonipa	;	as_Adlm_AC_fonipa
+aue_Adlm_AC_fonipa	;	ktz_Adlm_AC_fonipa
+ava_Adlm_AC_fonipa	;	av_Adlm_AC_fonipa
+ave_Adlm_AC_fonipa	;	ae_Adlm_AC_fonipa
+aym_Adlm_AC_fonipa	;	ay_Adlm_AC_fonipa
+ayr_Adlm_AC_fonipa	;	ay_Adlm_AC_fonipa
+ayx_Adlm_AC_fonipa	;	nun_Adlm_AC_fonipa
+aze_Adlm_AC_fonipa	;	az_Adlm_AC_fonipa
+azj_Adlm_AC_fonipa	;	az_Adlm_AC_fonipa
+bak_Adlm_AC_fonipa	;	ba_Adlm_AC_fonipa
+bam_Adlm_AC_fonipa	;	bm_Adlm_AC_fonipa
+baq_Adlm_AC_fonipa	;	eu_Adlm_AC_fonipa
+bcc_Adlm_AC_fonipa	;	bal_Adlm_AC_fonipa
+bcl_Adlm_AC_fonipa	;	bik_Adlm_AC_fonipa
+bel_Adlm_AC_fonipa	;	be_Adlm_AC_fonipa
+ben_Adlm_AC_fonipa	;	bn_Adlm_AC_fonipa
+bgm_Adlm_AC_fonipa	;	bcg_Adlm_AC_fonipa
+bh_Adlm_AC_fonipa	;	bho_Adlm_AC_fonipa
+bih_Adlm_AC_fonipa	;	bho_Adlm_AC_fonipa
+bis_Adlm_AC_fonipa	;	bi_Adlm_AC_fonipa
+bjd_Adlm_AC_fonipa	;	drl_Adlm_AC_fonipa
+bod_Adlm_AC_fonipa	;	bo_Adlm_AC_fonipa
+bos_Adlm_AC_fonipa	;	bs_Adlm_AC_fonipa
+bre_Adlm_AC_fonipa	;	br_Adlm_AC_fonipa
+bul_Adlm_AC_fonipa	;	bg_Adlm_AC_fonipa
+bur_Adlm_AC_fonipa	;	my_Adlm_AC_fonipa
+bxk_Adlm_AC_fonipa	;	luy_Adlm_AC_fonipa
+bxr_Adlm_AC_fonipa	;	bua_Adlm_AC_fonipa
+cat_Adlm_AC_fonipa	;	ca_Adlm_AC_fonipa
+ccq_Adlm_AC_fonipa	;	rki_Adlm_AC_fonipa
+cel_Adlm_AC_fonipa_gaulish	;	xtg_Adlm_AC_fonipa
```

Comment:
Also, could you explain to me why cel_Adlm_AC_fonipa_gaulish	will be canonicalized into xtg_Adlm_AC_fonipa ?

Reply:
Good catch. The code doesn't yet handle cases where items are split off with _x_. All the other cases are have type values that are not well-formed, so they aren't tested.